types/src/p2p.rs

```diff
@@ -0,0 +1,83 @@
+use crate::events::{l1::L1Event, l2::L2Event};
+use crate::{Hash, Slot, Transaction, UnixTimestamp};
+use borsh::{BorshDeserialize, BorshSerialize};
+use serde::{Deserialize, Serialize};
+use serde_with::{
+    base64::{Base64, Standard},
+    formats::Unpadded,
+};
+use std::mem::size_of;
+
+#[derive(Clone, Debug, PartialEq, Serialize, Deserialize)]
+pub struct SlotArtifact {
+    pub payload: SlotPayload,
+    pub events: Vec<L2Event>,
+}
```

Comment:
Lets rename it, as `artifact` is kinda meaningless word. You just get an idea that some computation has been done
```suggestion
pub struct SlotPayloadWithEvents {
    pub payload: SlotPayload,
    pub events: Vec<L2Event>,
}
```